itest/rust/src/transform3d_test.rs

```diff
@@ -0,0 +1,75 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+use crate::{assert_inner_equivalent, itest};
+
+use godot::prelude::{inner::InnerTransform3D, *};
+
+const TEST_TRANSFORM: Transform3D = Transform3D::new(
+    Basis::from_cols(
+        Vector3::new(1.0, 2.0, 3.0),
+        Vector3::new(4.0, 5.0, 6.0),
+        Vector3::new(7.0, 8.0, -9.0),
+    ),
+    Vector3::new(10.0, 11.0, 12.0),
+);
+
+pub(crate) fn run() -> bool {
+    let mut ok = true;
+    ok &= transform3d_equiv();
+    ok
+}
+
+#[itest]
+fn transform3d_equiv() {
+    assert_inner_equivalent!(InnerTransform3D, TEST_TRANSFORM.affine_inverse(), |a, b| {
+        Transform3D::is_equal_approx(&a, &b)
+    });
+    assert_inner_equivalent!(
+        InnerTransform3D,
+        TEST_TRANSFORM.orthonormalized(),
+        |a, b| { Transform3D::is_equal_approx(&a, &b) }
+    );
+    assert_inner_equivalent!(
+        InnerTransform3D,
+        TEST_TRANSFORM.rotated(Vector3::new(1.0, 2.0, 3.0).normalized(), 1.0),
+        |a, b| { Transform3D::is_equal_approx(&a, &b) }
+    );
+    assert_inner_equivalent!(
+        InnerTransform3D,
+        TEST_TRANSFORM.rotated_local(Vector3::new(1.0, 2.0, 3.0).normalized(), 1.0),
+        |a, b| { Transform3D::is_equal_approx(&a, &b) }
+    );
+    assert_inner_equivalent!(
+        InnerTransform3D,
+        TEST_TRANSFORM.scaled(Vector3::new(1.0, 2.0, 3.0)),
+        |a, b| { Transform3D::is_equal_approx(&a, &b) }
+    );
+    assert_inner_equivalent!(
+        InnerTransform3D,
+        TEST_TRANSFORM.scaled_local(Vector3::new(1.0, 2.0, 3.0)),
+        |a, b| { Transform3D::is_equal_approx(&a, &b) }
+    );
+    assert_inner_equivalent!(
+        InnerTransform3D,
+        TEST_TRANSFORM.translated_local(Vector3::new(1.0, 2.0, 3.0)),
+        |a, b| { Transform3D::is_equal_approx(&a, &b) }
+    );
+    assert_inner_equivalent!(
+        InnerTransform3D,
+        TEST_TRANSFORM.translated_local(Vector3::new(1.0, 2.0, 3.0)),
```

Comment:
Out of curiosity: Is it on purpose that e.g. `translated_local` is used twice (same in the 2d case, where also `rotated` was omitted)? From just briefly glancing over the code I haven't really understood where the "expected result" is actually coming from.

In any case great to see that you have adapted all the translated/scaled/rotated test cases from Godot already on the unit test level :+1: (I had added them to Godot as part of aligning their semantics.) So that aspect should be fully covered anyhow.

Reply:
Nope that was an accident.

These tests just check that calling the function through our rust-impl vs calling into godot does roughly the same thing. Basically by calling the provided function on `Transform3D` vs `InnerTransform3D`.